src/mpid/ch4/shm/posix/eager/iqueue/iqueue_init.c

```diff
@@ -56,12 +46,15 @@ static int init_transport(int vci_src, int vci_dst)
     transport->num_cells = MPIR_CVAR_CH4_SHM_POSIX_IQUEUE_NUM_CELLS;
     transport->size_of_cell = MPIR_CVAR_CH4_SHM_POSIX_IQUEUE_CELL_SIZE;
 
-    if (MPIR_CVAR_CH4_SHM_POSIX_IQUEUE_FREE_QUEUE_SENDER_SIDE) {
-        int queue_type = MPIDU_GENQ_SHMEM_QUEUE_TYPE__MPSC;
-        mpi_errno = MPIDU_genq_shmem_pool_create(transport->size_of_cell, transport->num_cells,
+    if (MPIR_CVAR_CH4_SHM_POSIX_TOPO_ENABLE) {
```

Comment:
Wait, so the else branch -- `MPIR_CVAR_CH4_SHM_POSIX_TOPO_ENABLE=0` -- will use `__MPMC` free queue?

Reply:
No. It is a typo. Will fix.